src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -132,50 +135,62 @@ String getClassNameInternal() {
     /**
      * Close the prepared statement's prepared handle.
      */
-    private void closePreparedHandle() {
+    private void closePreparedHandle()
+    {
         if (0 == prepStmtHandle)
             return;
 
         // If the connection is already closed, don't bother trying to close
-        // the prepared handle. We won't be able to, and it's already closed
+        // the prepared handle.  We won't be able to, and it's already closed
         // on the server anyway.
-        if (connection.isSessionUnAvailable()) {
+        if (connection.isSessionUnAvailable())
+        {
             if (getStatementLogger().isLoggable(java.util.logging.Level.FINER))
                 getStatementLogger().finer(this + ": Not closing PreparedHandle:" + prepStmtHandle + "; connection is already closed.");
         }
         else {
-            if (getStatementLogger().isLoggable(java.util.logging.Level.FINER))
-                getStatementLogger().finer(this + ": Closing PreparedHandle:" + prepStmtHandle);
+            // Using batched clean-up? If not, use old method of calling sp_unprepare.
+            if(1 < this.connection.getPreparedStatementDiscardActionThreshold()){
+                // Handle unprepare actions through batching @ connection level. 
+                this.connection.enqueuePreparedStatementDiscardItem(this.prepStmtHandle, this.executedSqlDirectly);
+                this.prepStmtHandle = 0;
+                this.connection.handlePreparedStatementDiscardActions(false);
+            }
+            else{
+                // Non batched behavior (same as pre batch impl.)
+                if (getStatementLogger().isLoggable(java.util.logging.Level.FINER))
```

Comment:
I looked through the rest of the code and this seems like the pattern to follow.